autobotAI_integrations/integrations/aws/__init__.py

```
@@ -197,13 +201,39 @@ def build_python_exec_combinations_hook(self, payload_task: PayloadTask,
             creds["aws_session_token"] = payload_task.creds.envs.get("AWS_SESSION_TOKEN")
+
+        # Long-running agent sessions (autobotAI deep agent) set a
+        # contextvar holding a refresh callable.  When present, build boto3
+        # clients with RefreshableCredentials so credentials auto-refresh
+        # per AWS API call — no ExpiredToken mid-call even if a single
+        # python_sdk task runs longer than the STS TTL.  Lambda/CLI callers
+        # leave the contextvar unset and get the existing static-creds path.
+        resolver = current_aws_creds_resolver.get()
+        use_refreshable = resolver is not None
+
+        def _global_client(name: str):
```

Ruff (0.15.14)

[warning] Missing return type annotation for private function `_global_client`

(ANN202)